synapse/storage/_base.py

```diff
@@ -76,14 +76,16 @@ def _attempt_to_invalidate_cache(
         """
 
         try:
-            if key is None:
-                getattr(self, cache_name).invalidate_all()
```

Comment:
I'm changing this because the fact that the `try/except` wrapped more than the `getattr` meant that an important error got swallowed.